tests/functional/test_endpoints.py

```diff
@@ -92,15 +144,22 @@ def test_service_name_matches_endpoint_prefix():
 
 
 def _assert_service_name_matches_endpoint_prefix(loader, service_name):
-    # Load the service model and grab its endpoint prefix
     service_model = loader.load_service_model(service_name, 'service-2')
-    endpoint_prefix = service_model['metadata']['endpointPrefix']
+    computed_name = _get_computed_service_name(service_model['metadata'])
 
     # Handle known exceptions where we have renamed the service directory
     # for one reason or another.
-    expected_endpoint_prefix = SERVICE_RENAMES.get(service_name, service_name)
+    actual_service_name = SERVICE_RENAMES.get(service_name, service_name)
     assert_equal(
-        endpoint_prefix, expected_endpoint_prefix,
-        "Service name `%s` does not match expected endpoint "
-        "prefix `%s`, actual: `%s`" % (
-            service_name, expected_endpoint_prefix, endpoint_prefix))
+        computed_name, actual_service_name,
+        "Actual service name `%s` does not match expected service name "
+        "we computed: `%s`" % (
+            actual_service_name, computed_name))
+
+
+def _get_computed_service_name(service_metadata):
+    if 'serviceId' not in service_metadata:
```

Comment:
So it is unfortunate that we have not rolled all of the service id's out yet. However, I am a little hesitant to continue to use endpointPrefix because this test may all of a sudden fail if we pull in a service model that now has a service id. For example let's take `resourcegroupstaggingapi`. Right now its endpoint prefix is `tagging` and that override is hard coded. However when its service id gets added to the model, it will be `Resource Groups Tagging API` which translates to `resoruce-groups-tagging-api` which will cause the override to not happen and the test to fail. I'm not sure how many examples there are of this.

I think what may make sense is to try to compute the service id if it does not exist and update the override dictionary . To my knowledge, (@stealthycoin can confirm)  all of the service id's to be rolled are based off of the computed form so there should be no discrepancies.

Reply:
Yep!

Reply:
Ok chatted offline and wanted to capture what we'd discuss.  In order to address this concern, I'll just update the PR to add all the missing service ids.  I'll also make a missing service id a a test failure.  That should address all the concerns here.